src/transaction.js

```diff
@@ -190,6 +201,107 @@ function signatureForm (
   return newTx
 }
 
+function deserializeCertFields (buf: Buffer, offset: number) {
+  const sidechainId = buf.slice(offset, offset + 32).reverse().toString('hex');
+  offset += 32
+
+  const epochNumber = buf.readInt32LE(offset);
+  offset += 4 
+
+  const quality = zbufferutils.readUInt64LE(buf, offset)
+  offset += 8; 
+
+  const endEpochCumScTxCommTreeRootLen = varuint.decode(buf, offset);
+  offset += varuint.decode.bytes
+  const endEpochCumScTxCommTreeRoot = buf.slice(offset, offset + endEpochCumScTxCommTreeRootLen).toString('hex');
+  offset += endEpochCumScTxCommTreeRootLen
+
+  const scProofLen = varuint.decode(buf, offset);
+  offset += varuint.decode.bytes
+  const scProof = buf.slice(offset, offset + scProofLen).toString('hex');
+  offset += scProofLen
+
+  const vFieldElementCertificateFieldLen = varuint.decode(buf, offset);
+  offset += varuint.decode.bytes
+  let vFieldElementCertificateField = [];
+  for (let i = 0; i < vFieldElementCertificateFieldLen; i++) {
+    let fieldLen = varuint.decode(buf, offset);
+    offset += varuint.decode.bytes
+
+    vFieldElementCertificateField.push(buf.slice(offset, offset + fieldLen).toString('hex'));
+    offset += fieldLen
+  }
+
+  const vBitVectorCertificateFieldLen = varuint.decode(buf, offset);
+  offset += varuint.decode.bytes
+  let vBitVectorCertificateField = [];
+  for (let i = 0; i < vBitVectorCertificateFieldLen; i++) {
+    let fieldLen = varuint.decode(buf, offset);
+    offset += varuint.decode.bytes
+
+    vBitVectorCertificateField.push(buf.slice(offset, offset + fieldLen).toString('hex'));
+    offset += fieldLen
+  }
+
+  const ftScFee = zbufferutils.readUInt64LE(buf, offset);
+  offset += 8;
+  
+  const mbtrScFee = zbufferutils.readUInt64LE(buf, offset)
+  offset += 8;
+  
+  const cert = {
+    sidechainId,
+    epochNumber,
+    quality,
+    endEpochCumScTxCommTreeRoot,
+    scProof,
+    vFieldElementCertificateField,
+    vBitVectorCertificateField,
+    ftScFee,
+    mbtrScFee,
+  }
+
+  return [cert, offset]
```

Comment:
I don't believe we need to return the offset because the parent functions still have the reference in scope

Reply:
I think that's only the case if I defined `deserializeCertFields` inside `deserializeTx` and didn't pass `offset` in as an argument to the function.

Reply:
Oh I see what you're doing now, yep that makes sense